### DISCUSSION TITLE
Something like atom composition in Jotai?

### DISCUSSION THREAD

**Discussion by mattp0123** [pmndrs/zustand#1058] [Q&A] [answered]
When using Jotai, we can do this:
```typescript
const numAtom = atom(0)
const numPlus1Atom = atom((get) => get(numAtom) + 1)
```
This is super useful in situation like:
```typescript
const todosAtom = atom<Todo[]>([])
const doneTodos = atom(get => get(todosAtom).filter(todo => todo.isDone))
const undoneTodos = atom(get => get(todosAtom).filter(todo => todo.isDone === false))
```
The filter calculation will only happen once unless the todosAtom is changing, even if I use these atoms in many different components. But in Zustand, I can only do this:
```typescript
function useDoneTodos() {
  const todos = useStore(state => state.todos)
  const doneTodos = useMemo(() => todos.filter(todo => todo.isDone), [todos])
  return doneTodos
}
```
^ The filter calculation will happen n times if I put this hook in n components

Is there any more performant way to do this? Maybe I have to use the Context API? Or migrate to Jotai 😂

**Comment by dai-shi** [comment, marked as answer]
In such cases, you want to use memoized selectors with such as reselect and [proxy-memoize](https://github.com/dai-shi/proxy-memoize).

Redux docs should help! 😁 
https://redux.js.org/usage/deriving-data-selectors
